src/middlewared/middlewared/plugins/webdav.py

```diff
@@ -153,19 +153,29 @@ class Config:
     async def upper(self, data):
         data['protocol'] = data['protocol'].upper()
         data['htauth'] = data['htauth'].upper()
+        if data['certssl']:
+            data['certssl'] = await self.middleware.call(
+                'certificate.query',
+                [['id', '=', data['certssl']['id']]],
+                {'get': True}
+            )
 
         return data
 
     @private
     async def validate(self, data, schema_name):
         verrors = ValidationErrors()
 
-        if (data.get('protocol') == 'httphttps' and data.get(
-                'tcpport') == data.get('tcpportssl')):
-            verrors.add(f"{schema_name}.tcpportssl",
-                        'The HTTP and HTTPS ports cannot be the same!')
+        if data.get('protocol') == 'httphttps' and data.get('tcpport') == data.get('tcpportssl'):
+            verrors.add(
+                f"{schema_name}.tcpportssl",
+                'The HTTP and HTTPS ports cannot be the same!'
+            )
 
-        if (data.get('protocol') != 'http' and data.get('certssl') is None):
+        cert_ssl = data.get('certssl') or 0
+        if data.get('protocol') != 'http' and not (
+                await self.middleware.call('certificate.query', [['id', '=', cert_ssl]])
+        ):
```

Comment:
Not choosing a certificate and it not existing are distinct things, we should probably note that or the error message will be misleading.